### PR TITLE
Fixed #5402, #5502: fixed Keybinding widget UI issues

### LINKED ISSUE #5402: [keybindings] hidden key icons on small browser window
When the browser window becomes small, some key icons in `Keybindings` are automatically hidden/incomplete. From the user experience perspective, it would be better to prioritize hiding `Commands` on the right.

theia-keybindings-resize

### PR DESCRIPTION
**Pull request by fangnx**
Fixed #5402, #5502.

- ~~Made `Label` and `Keybinding` columns fixed width so that their contents are always displayed properly on resize.~~
- Fixed the display of the edit and reset icons.

### CODE REVIEW

**Comment by vince-fugnitto** [comment]
It looks much better!
Do you think we should keep the `ellipsis` for the other columns?
(scope, context and command)

<div align='center'>

Screen Shot 2019-06-19 at 11 37 57 AM


</div>

**Comment by vince-fugnitto** [comment]
It feels a little odd to me that something like this can happen.

<div align='center'>

Screen Shot 2019-06-19 at 11 40 58 AM


</div>

**Comment by fangnx** [comment]
> It looks much better!
> Do you think we should keep the `ellipsis` for the other columns?
> (scope, context and command)
> 
> Screen Shot 2019-06-19 at 11 37 57 AM

Somehow keeping text ellipsis causes some display issue when resized. I will try again and maybe we should keep it to be consistent with other widgets.

**Comment by lmcbout** [comment]
Fix column is not necessary the right thing to do, did you take into account the font size! Also, some screens have different resolution

**Comment by fangnx** [comment]
I 

> It feels a little odd to me that something like this can happen.
> 
> Screen Shot 2019-06-19 at 11 40 58 AM
> In comparison, here's how vscode maximizes the available space:
> 
> Screen Shot 2019-06-19 at 11 43 07 AM

I think the idea is to set the width to be the widest cell in the `keybinding` column?

**Comment by vince-fugnitto** [comment]
> I guess what should be done is to set the width to be the widest cell in the `keybinding` column?

From the looks of it, each column is present (even on small windows) but the keybindings themselves have highest priority (they are hidden last). Every other column has `text-overflow: ellipsis` for smaller viewports. 

I'm not sure what the best look would be for smaller viewports, do you have any suggestions?

**Comment by fangnx** [comment]
> > I guess what should be done is to set the width to be the widest cell in the `keybinding` column?
> 
> From the looks of it, each column is present (even on small windows) but the keybindings themselves have highest priority (they are hidden last). Every other column has `text-overflow: ellipsis` for smaller viewports.
> 
> I'm not sure what the best look would be for smaller viewports, do you have any suggestions?

I tried to resize the window for VS Code, but I did not see the keybinding having the highest priority. All columns seem to be squeezed together.

image

**Comment by vince-fugnitto** [comment]
> I tried to resize the window for VS Code, but I did not see the keybinding having the highest priority. All columns seem to be squeezed together.

At a certain point yes, but before that every other column displays `ellipsis` (that is what I mean about highest priority, it is the last to resize).

In the screenshot you can see that `label` and `when` have `ellipsis` before keybindings are displayed oddly.
<div align='center'>

Screen Shot 2019-06-19 at 1 15 28 PM


</div>

<br />

Also, I'm not arguing that vscode displays the keybindings the best (as you can see from your screenshot), therefore we can implement something better.

**Comment by fangnx** [comment]
> > I tried to resize the window for VS Code, but I did not see the keybinding having the highest priority. All columns seem to be squeezed together.
> 
> At a certain point yes, but before that every other column displays `ellipsis` (that is what I mean about highest priority, it is the last to resize).
> 
> In the screenshot you can see that `label` and `when` have `ellipsis` before keybindings are displayed oddly.
> 
> Screen Shot 2019-06-19 at 1 15 28 PM
> 
> Also, I'm not arguing that vscode displays the keybindings the best (as you can see from your screenshot), therefore we can implement something better.

Do you have any advice/suggestions? I am not quite sure what it is the best to implement ;)

**Comment by vince-fugnitto** [comment]
> Do you have any advice/suggestions? I am not quite sure what it is the best to implement ;)

All I can really think of is to try and maximize the available space at the moment:
1. each column is left aligned and by default the maximum column width completely displays the values.
2. make sure every column wraps (ellipsis) besides `keybindings` and `actions`.
3. try to keep as much the space available for the `actions` and `keybindings` to display fully.
4. we have tooltips for each column (so it's ok to not display other columns fully).

I think that might be enough, we want to improve the display, doesn't mean it'd be perfect in the end.

**Comment by fangnx** [comment]
> > Do you have any advice/suggestions? I am not quite sure what it is the best to implement ;)
> 
> All I can really think of is to try and maximize the available space at the moment:
> 
> 1. each column is left aligned and by default the maximum column width completely displays the values.
> 2. make sure every column wraps (ellipsis) besides `keybindings` and `actions`.
> 3. try to keep as much the space available for the `actions` and `keybindings` to display fully.
> 4. we have tooltips for each column (so it's ok to not display other columns fully).
> 
> I think that might be enough, we want to improve the display, doesn't mean it'd be perfect in the end.

Nice points! Though it seems inevitable that we "set" the Keybinding column to some width in order for it to display fully (if we want it to).

**Comment by vince-fugnitto** [comment]
> Nice points! Though it seems inevitable that we "set" the Keybinding column to some width in order for it to display fully (if we want it to).

I'm not sure, all other columns would have a `text-overflow` and `wrap` assigned to them, while the `keybindings` wouldn't, maybe that'd be enough or at least a starting point?

File: packages/keymaps/src/browser/keybindings-widget.tsx [hunk 1]
```diff
@@ -255,12 +255,12 @@ export class KeybindingWidget extends ReactWidget {
                 <table>
                     <thead>
                         <tr>
-                            <th className='th-action'></th>
-                            <th className='th-label'>Label</th>
-                            <th className='th-keybinding'>Keybinding</th>
-                            <th className='th-scope'>Scope</th>
-                            <th className='th-context'>Context</th>
-                            <th className='th-command'>Command</th>
+                            <th className='th-action kb-action'></th>
```
Comment (vince-fugnitto) [inline packages/keymaps/src/browser/keybindings-widget.tsx]: I don't think these new classes are necessary.

File: packages/keymaps/src/browser/keybindings-widget.tsx [hunk 2]
```diff
@@ -276,24 +276,28 @@ export class KeybindingWidget extends ReactWidget {
             {
                 this.items.map((item, index) =>
                     <tr className='kb-item-row' key={index} onDoubleClick={a => this.editKeybinding(item)}>
-                        <td className='kb-actions'>{this.renderActions(item)}</td>
-                        <td title={this.getRawValue(item.command)}>{this.renderMatchedData(item.command)}</td>
-                        <td title={(item.keybinding) ? this.getRawValue(item.keybinding) : ''} className='monaco-keybinding'>
+                        <td className='kb-action'>{this.renderActions(item)}</td>
```
Comment (vince-fugnitto) [inline packages/keymaps/src/browser/keybindings-widget.tsx]: Why is the rename `kb-actions` to `kb-action` necessary?

**Pull request merged**